src/porepy/geometry/distances.py

```diff
@@ -607,21 +654,33 @@ def segment_overlap_segment_set(
     The function currently works only for 2D geometries: ``nd == 2``.
 
     Parameters:
-        start (shape=(nd,)): Start point of a segment.
-        end (shape=(nd,)): End point of a segment.
-        start_set (shape=(nd, num_segments)): Start points of the set of segments.
-        end_set (shape=(nd, num_segments)): End points of the set of segments.
-        return_indices (optional): Whether the function should also return
-          indices of overlappings.
-        tol (optional): Tolerance to be used.
+        start: ``shape=(nd,)``
+
+            Start point of a segment.
+        end: ``shape=(nd,)``
+
+            End point of a segment.
+        start_set: ``shape=(nd, num_segments)``
+
+            Start points of the set of segments.
+        end_set: ``shape=(nd, num_segments)``
+
+            End points of the set of segments.
+        return_indices: ``default=False``
+
+            Whether the function should also return indices of overlappings.
+        tol: ``default=1e-5``
+
+            Tolerance to be used.
 
     Returns:
         If ``return_indices==True``, we return a tuple of 2 elements.
 
-        bool:
+        :obj:`bool`:
             ``True`` if the segment overlaps any of the segments in the set, ``False``
             otherwise.
-        ndarray ``(shape=(num_overlapping,))``:
+        :obj:`~numpy.ndarray`: ``shape=(num_overlapping,)``
+
```

Comment:
Have I understood it correctly that you need the blank between `:obj:...` and the description in cases where there is a shape argument, but not otherwise?

Reply:
Yes, shape, default or any other additional specifications. The blank/ linea break gives a readable format which distinguishes between the format specification and the content/arg description.

Without the blank I have noticed that Sphinx produces some not-so-nice-looking content containers, especially if the text after the specifications is short and no explicit line break is introduced.

Reply:
Thanks.